torch/_refs/nn/functional/__init__.py

```diff
@@ -357,6 +390,21 @@ def l1_loss(
     return _apply_loss_reduction(loss, reduction)
 
 
+# CompositeImplicitAutograd - don't register decomp
+def log_softmax(
+    a: TensorLikeType,
+    dim: Optional[int] = None,
+    _stacklevel: int = 3,  # for compat when using TorchRefsMode(strict=True)
+    dtype: Optional[torch.dtype] = None,
+) -> TensorLikeType:
+    # The error is for compat with regular PyTorch, which has this behavior
+    # deprecated.  For PrimTorch, it's fine to drop support for deprecated
+    # behavior because it requires explicit opt in.  This error is to inform
+    # users how to update their calls.
+    check(dim is not None, lambda: "implicit dim not supported, use dim=X")
+    return torch.log_softmax(a=a, dim=dim, dtype=dtype)  # type: ignore[call-overload]
```

Comment:
Same concern about `log_softmax` -- do we expect it to be prim? I don't think so. Shouldn't these references implement the algorithms in a numerically accurate way?

Reply:
The refs are implemented in the `torch.` namespace. These refs deal with those pesky `_stacklevel` args. See https://github.com/pytorch/pytorch/pull/84502#discussion_r963578734